activerecord/lib/active_record/result.rb

```diff
@@ -130,6 +130,14 @@ def cast_values(type_overrides = {}) # :nodoc:
       end
     end
 
+    def column_type(name, index, type_overrides)
```

Comment:
I someone has an idea to retrieve the correct type without spreading the fallback logic all over the place and making this method public, hints would be very welcome.

Reply:
I'm not sure it answers your question directly, but you should remove this from the public API.

```suggestion
    def column_type(name, index, type_overrides) # :nodoc:
```